File: frontend/src/types/problem.ts
export interface Problem {
  contestId?: number;
  problemsetName?: string;
  index: string;
  name: string;
  type: string;
  rating?: number;
  tags: string[];
  solved_count?: number;
}

export interface ProblemFilters {
  tags?: string;
  min_rating?: number;
  max_rating?: number;
  search?: string;
  sort_by?: string;
  limit?: number;
  offset?: number;
}

export interface ProblemsResponse {
  problems: Problem[];
  total: number;
  limit: number;
  offset: number;
}

File: backend/app/models/problem.py
from pydantic import BaseModel
from typing import List, Optional


class Problem(BaseModel):
    """Codeforces problem model"""
    contestId: Optional[int] = None
    problemsetName: Optional[str] = None
    index: str
    name: str
    type: str
    rating: Optional[int] = None
    tags: List[str] = []

    class Config:
        from_attributes = True


class ProblemStatistics(BaseModel):
    """Problem statistics model"""
    contestId: Optional[int] = None
    index: str
    solvedCount: int

    class Config:
        from_attributes = True


class ProblemWithStats(BaseModel):
    """Problem combined with statistics"""
    problem: Problem
    statistics: Optional[ProblemStatistics] = None
